lib/formSchemaGenerator.ts

```
@@ -0,0 +1,261 @@
+
+    if (value === null || value === undefined) {
+      // Use default value if data is null/undefined
+      const defaults = generateDefaultValues([field]);
+      transformedData[fieldKey] = defaults[fieldKey];
+      return;
+    }
+
+    switch (field.fieldType) {
+      case 'number':
+        // Ensure number fields are actually numbers
+        const numValue = typeof value === 'string' ? parseFloat(value) : value;
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)